cmd/agola/cmd/runlist.go

```diff
@@ -60,11 +66,24 @@ func init() {
 	cmdRun.AddCommand(cmdRunList)
 }
 
-func printRuns(runs []*gwapitypes.RunResponse) {
+func printRuns(runs []*runDetails) {
 	for _, run := range runs {
-		fmt.Printf("%s: Phase: %s, Result: %s\n", run.ID, run.Phase, run.Result)
-		for _, task := range run.Tasks {
+		fmt.Printf("%s: Phase: %s, Result: %s\n", run.runResponse.ID, run.runResponse.Phase, run.runResponse.Result)
+		i := 0
+		for _, task := range run.runResponse.Tasks {
```

Comment:
just use 
```
for i, task := range run.runResponse.Tasks
```

and remove the `i++`

Reply:
In this way `i` will be the key not the index (`run.runTaskResponse` is a `map[string]*RunResponseTask`)

Reply:
It's the run.runTaskResponse array not the run.Tasks map. Using the map is wrong since the order is random so the index won't match the correct task from the array.